service/matching/physical_task_queue_manager.go

```diff
@@ -423,7 +423,18 @@ func (c *physicalTaskQueueManagerImpl) DispatchNexusTask(
 	request *matchingservice.DispatchNexusTaskRequest,
 ) (*matchingservice.DispatchNexusTaskResponse, error) {
 	deadline, _ := ctx.Deadline() // If not set by user, our client will set a default.
-	task := newInternalNexusTask(taskId, deadline, request)
+	var opDeadline time.Time
+	opTimeoutHeader, set := request.GetRequest().GetHeader()["Operation-Timeout"]
```

Comment:
What if header is `nil` here? What if the header is user provided (e.g. via curl) and is in lower case?

Reply:
Also note that this is only applicable for `StartOperation` requests but I don't think it hurts to manipulate on the other requests.